sftp/sftp.go

```diff
@@ -0,0 +1,97 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// PasswordAuth indicates password-based authentication
+	PasswordAuth = "passwordAuth"
+	// KeyAuth indicates key-based authentication
+	KeyAuth = "keyAuth"
+)
+
+// SFTPManager interface abstracts the SFTP client
+type SFTPManager interface {
+	UploadFile(localFilePath, remoteDir string) error
+	DownloadFile(remoteFilePath, localDir string) error
+	DeleteFile(remoteFilePath string) error
+}
+
+// SFTPManagerImpl is a real implementation of SFTPManager
+type SFTPManagerImpl struct {
+	client SFTPClient
+}
+
+func NewSFTPManager(sshClient *ssh.Client) (SFTPManager, error) {
+	sftpClient, err := NewSFTPClient(sshClient)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create SFTP client: %w", err)
+	}
+	return &SFTPManagerImpl{client: sftpClient}, nil
+}
+
+// UploadFile uploads a file to the remote server
+func (r *SFTPManagerImpl) UploadFile(localFilePath, remoteDir string) error {
+	localFile, err := os.Open(localFilePath)
+	if err != nil {
+		return err
+	}
+	defer localFile.Close()
+
+	remoteFileName := filepath.Join(remoteDir, filepath.Base(localFilePath))
+	remoteFile, err := r.client.Create(remoteFileName)
+	if err != nil {
+		return err
+	}
+	defer remoteFile.Close()
+
+	_, err = io.Copy(remoteFile, localFile)
+	if err != nil {
+		fmt.Printf("Error copying file: %v\n", err)
+		return err
```

Comment:
We cannot have these. I suggest not logging since the caller can log this, it seems unnecessary to use a logger in this layer. Otherwise if you really want to log then I suggest using a logger with a configurable log level like the [go-kit logger](https://github.com/rudderlabs/rudder-go-kit/blob/v0.23.3/logger/logger.go#L48).

```suggestion
		return fmt.Errorf("error copying file: %w", err)
```